### PR TITLE
[IMP] web, web_editor: use system-fonts

### PR DESCRIPTION
**Pull request by stefanorigano**
Use system-fonts for backend and web-editor's UI.
This commit removes references to proprietary fonts and allows the OS
to use the system ones. The aim is to reduce http request and the
general footprint (especially for enterprise, https://github.com/odoo/enterprise/pull/31529).

The new font-stack:

```
- apple-system (San Francisco): iOS Safari, macOS Safari, macOS Firefox
- BlinkMacSystemFont (San Francisco): macOS Chrome
- Segoe UI: Windows
- Helvetica Neue: old OSX versions
- Roboto: Android, Chrome OS
- Ubuntu: Ubuntu
- Liberation Sans: Linux (others)
- Arial: Any
- sans-serif: General fallback

```
enterprise:
- https://github.com/odoo/enterprise/pull/31529

task-2995248

### CODE REVIEW

**Comment by stefanorigano** [comment]
Hello @qsm-odoo and @dmo-odoo ,  backend looks fine(-ish depending by the OS, but it's something we're apparently ok with), any blocking point on your sides?

File: addons/website/static/src/scss/primary_variables.scss [hunk 1]
```diff
@@ -2028,13 +2028,7 @@ $o-font-aliases-to-keys: (
...
     'buttons': 'buttons-font',
 );
-$o-website-values-palettes: (
-    (
-        'headings-font': 'Source Sans Pro',
-        'navbar-font': 'Source Sans Pro',
-        'buttons-font': 'Source Sans Pro',
-    ),
-) !default;
+$o-website-values-palettes: () !default;
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/primary_variables.scss:2031]: @stefanorigano Doing this is far from enough to use system fonts (if I am not mistaken, doing this you will just use Roboto (loaded from google) for the whole website as the first font defined in the website map will be used). I have already done a PR to use system fonts for the website here: https://github.com/odoo/odoo/pull/100867 so maybe remove this website change and focus on the backend only :+1: Thanks :+1:

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/src/scss/primary_variables.scss (hunk 1 above)
Comment (stefanorigano) [reply addons/website/static/src/scss/primary_variables.scss:2031]: Oh... I knew you made some tests but I didn't know you had already a PR 👍 

Should we define `$o-system-font-stack` globally? Or you prefer to keep it under control on website?

Comment (qsm-odoo) [reply addons/website/static/src/scss/primary_variables.scss:2031]: Mmh, I guess we could indeed. Although, you would still need to do `$o-font-family-sans-serif: o-add-unicode-support-font($o-system-font-stack);` (because the unicode support is automatically added in website fonts later so I cannot put it directly when I defined the variable). Should also rename it `$o-system-fonts` to be consistent with the rest of my PR actually.

Ok: I'll move the variable in web in my PR (RDE will soon merge it, like monday maximum) then you'll be able to rebase your branch afterwards to use the shared variable in the backend, seems ok to you ? :+1:

Comment (stefanorigano) [reply addons/website/static/src/scss/primary_variables.scss]: If you plan to stay strict to BS is probably better to define a variable for the frontend actually:
- `Arial` is not good for the backend; semibold is missing and we need it for our 13px buttons that looks too bulky when using **bold**; so I moved it at the end of the list
- We want Ubuntu font to match that OS
-  I've read weird things about the `system-ui`, but maybe were just old news , I should check

Comment (qsm-odoo) [reply addons/website/static/src/scss/primary_variables.scss]: No, I think it's better to share. If you fix/improve the font system stack for the backend, it probably make sense for the website too. As those are fonts that are supposed to be in the system, I don't see a change you would want to the backend and not the website.

I am changing my PR using the same font stack as bootstrap, once it is merged, you'll be free to modify the font stack as you wish. I am also fixing stuff I missed:
- I forgot to use `o-add-unicode-support-font` on the "theme misconfiguration fallback" I made (thought of it explaining it to you so thanks :p)
- When website is not installed, the frontend use the bootstrap font stack... but without the unicode support of Odoo (I guess this could be a fix in stable, but let's merge it for 16.0 in my branch as a first step, not sure it is worth changing in stable).
- The `o-add-unicode-support-font` actually does not work the right way on the font stack of bootstrap. It puts the unicode support in between emoji fonts while it should be before serif/sans-serif (could also be a fix in stable but there is no broken case in stable since we don't use that font stack).

Comment (qsm-odoo) [reply addons/website/static/src/scss/primary_variables.scss]: => https://github.com/odoo/odoo/pull/100867#issuecomment-1256483248

Once RDE has merged that PR, you can thus rebase this one and use the variable I defined in web and you can modify it in whatever way you like :+1:

Comment (stefanorigano) [reply addons/website/static/src/scss/primary_variables.scss]: Perfect, thanks 👍

**Review by adr-odoo — APPROVED** [review]
@robodoo r+

**Comment by qsm-odoo** [comment]
@robodoo r-

Loading the homepage of the website as a public user:

Before this PR:
image

After this PR:
image

**Comment by stefanorigano** [comment]
> Loading the homepage of the website as a public user:

@qsm-odoo, I get fontAwesome only in both Community and Enterprise (public or logged-in).
Tested on Chrome and Firefox (OSX) and Edge (windows).
 
I'm puzzled, do you have any insight? 

image

**Comment by qsm-odoo** [comment]
Try removing everything which is before "Roboto" in the font stack and you should have the same thing as me in all browsers and OS I guess? In any case, there must an inclusion of Roboto somewhere in our code that can now be gone? At minimum, on your branch, I have this in my folders:
image
Probably no more reason to have them and probably there is code still including them that has to be removed?

**Comment by stefanorigano** [comment]
> Try removing everything which is before "Roboto" in the font stack [...]

@qsm-odoo , yes, Roboto shows up in that case... but in `16.0` too :/

**16.0** (homepage, public user)
`$o-system-fonts: ( Roboto, "Helvetica Neue", Arial, "Noto Sans", "Liberation Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji") !default;`

image

**Comment by qsm-odoo** [comment]
Then it was an error before too but which now always shows up on Ubuntu with this PR :wink: The point is that the repo Roboto fonts should not be included in the repo anymore (and the code which includes them)... any reason to not remove it? :thinking:

**Comment by stefanorigano** [comment]
Can't be removed, it's used by reports, and it's probably the issue.

Report's fonts are injected in common_reports bundle, but also in the backend and fronted one ;)

Did some tests, seems to work, I'll push once done

**Review by adr-odoo — APPROVED** [review]
LGTM but I will wait qsm review

**Review by qsm-odoo — COMMENTED** [review]
I push-forced both branches to review the commit messages (and did a rebase without change at the same time) (so don't forget to fetch if you were to modify this again):
- The mentioned font stack was not the one used in the commit
- IMP instead of REF
- Fix apps lists
- Mention the commit introducing font stack in frontend + the fix this provides
- ...

I went and test, it looks good :+1: Although some concerns, I let you decide if you want to ignore them, treat them in another PR later or do it before r+ of this one:

- I see this code in spreadsheet, seems strange:
```javascript
const DEFAULT_FONT = "'Roboto', arial";
```

- Point of sale still use repo fonts instead of system fonts (found it grepping `web/static/fonts/fonts.scss`). Should not it be aligned to the rest?

-  On top of point of sale, I went and grep "Lato" in the repo... there are other occurences than point of sale: barcodes_mobile, event_barcode, mrp_workorder, ... should not all of these be converted too? (I guess we can't remove the Lato fonts entirely though as seems to be used in reports (but maybe their definition could be simplified))

- Don't know what it is but there are many instances of "Roboto" in /debian, maybe worth looking if that has to stay (maybe ask to the one who introduced those lines?)

**Comment by rrahir** [comment]
> Thanks @qsm-odoo 👍
> 
> > I see this code in spreadsheet, seems strange: `const DEFAULT_FONT = "'Roboto', arial";`
> 
> yeah, it's in in library, we "can't" change it I guess. Android devices will use _Roboto_ and the others will fallback to _Arial_. I would be nice if spreadsheet uses the same font as the one the user renders when using the the backend though. @rrahir , do you think that we can change the lib to match the new font-stack once merged?
 
We can definitely try it out !  We could have this default font overridable but if I understand correctly, we chose to give more freedom the browser and we need to see how this can be translated to the canvas, I don't really know if it defaults to the browser font nor how the lib would look with it, I know @bouvyd raised some concerns about the lack of boldness in some cases...

Anyway, we can look into it :)

**Comment by qsm-odoo** [comment]
**TL;DR:** You can r+ when you wish @adr-odoo 

--

> It will not affect backend/frontend 'cause it's in point_of_sale.assets

Yep, like I said, nothing which prevents to r+ this anymore :wink: All points I made can be done later if they are even relevant :+1: 

> These modules are already not compliant with the UI's main font. I guess that there is a reason if each of these modules enforce a specific font...

Maybe indeed, I guess I lack faith :smile: 

> The PR aims to change the default font, not investigate and solve all font-related issues across modules :/

Like I said, I let you all judge :wink: I have no time to fix all the modules either (but to be realistic: nobody will fix fonts that work if not in this kind of PR) :wink: 

> Consider that we still include these fonts and use them in reports

True, my last point was probably useless, I was merely curious about what those lines are (like I said, no blocking point).

**Comment by adr-odoo** [comment]
@robodoo r+
